kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kms/aws/AwsKmsLocal.java

```diff
@@ -0,0 +1,184 @@
+ */
+
+package io.kroxylicious.systemtests.installation.kms.aws;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.ServicePort;
+
+import io.kroxylicious.kms.provider.aws.kms.AwsKmsTestKmsFacade;
+import io.kroxylicious.systemtests.Environment;
+import io.kroxylicious.systemtests.k8s.exception.KubeClusterException;
+import io.kroxylicious.systemtests.resources.manager.ResourceManager;
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+import io.kroxylicious.systemtests.utils.TestUtils;
+import io.kroxylicious.systemtests.utils.VersionComparator;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * The type Aws kms local.
+ */
+public class AwsKmsLocal implements AwsKmsClient {
+    public static final String LOCALSTACK_SERVICE_NAME = "localstack";
+    public static final String LOCALSTACK_DEFAULT_NAMESPACE = "localstack";
+    public static final String LOCALSTACK_HELM_REPOSITORY_URL = "https://localstack.github.io/helm-charts";
+    public static final String LOCALSTACK_HELM_REPOSITORY_NAME = "localstack";
+    public static final String LOCALSTACK_HELM_CHART_NAME = "localstack/localstack";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsKmsLocal.class);
+    private static final String AWS_LOCAL_CMD = "awslocal";
+    private final String deploymentNamespace;
+    private String podName;
+    private String installedLocalStackVersion;
+    private String region;
+
+    /**
+     * Instantiates a new Aws.
+     *
+     */
+    public AwsKmsLocal() {
+        this.deploymentNamespace = LOCALSTACK_DEFAULT_NAMESPACE;
+    }
+
+    @Override
+    public String getAwsCmd() {
+        return AWS_LOCAL_CMD;
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return Environment.AWS_ACCESS_KEY_ID.equals(Environment.AWS_ACCESS_KEY_ID_DEFAULT);
+    }
+
+    /**
+     * Gets the installed version.
+     *
+     * @return the version
+     */
+    public String getLocalStackVersionInstalled() {
+        if (installedLocalStackVersion == null) {
+            URI url = getAwsUrl();
+            try (var output = new ByteArrayOutputStream();
+                    var error = new ByteArrayOutputStream();
+                    var exec = kubeClient().getClient().pods()
+                            .inNamespace(deploymentNamespace)
+                            .withName(podName)
+                            .writingOutput(output)
+                            .writingError(error)
+                            .exec("sh", "-c", "curl " + url + "/_" + LOCALSTACK_SERVICE_NAME + "/info")) {
```

Comment:
Use a java HttpClient

Reply:
This was intended to be run inside the kubernetes cluster, but I will test if we can access from outside to `info`

Reply:
Changed. Thanks!